File: 2020/1.rb
lines = File.open('1.txt').readlines.map(&:strip).map(&:to_i)
lines.each do |first|
	lines.each do |second|
		if first + second  == 2020
			puts first * second 
		end
		lines.each do |third|
			if first + second + third == 2020
				puts first * second * third
			end
		end
	end
end

#Who cares if it's brute force ... it didn't take very long to write
#better solution in python:
#lines = list(map(int, open('1.txt', r).readlines()))
#for i in range(len(lines)):
#    for j in range(i + 1, len(lines)):
#        print (lines[i] * lines[j]) if lines[i] + lines[j] == 2020
#        for k in range(j + 1, len(lines)):
#            print (lines[i] * lines[j] * lines[k]) if lines[i] + lines[j] + lines[k]== 2020

File: 2020/1.py
lines = list(map(int,map(str.strip, open('1.txt', 'r').readlines())))

twoFound, threeFound = False, False
for i in lines:
    for j in lines:
        for k in lines:
            if i + j == 2020 and not twoFound:
                # part 1: brute force :)
                print("Part 1: " +  str(i*j))
                twoFound = True
            if i + j + k == 2020 and not threeFound:
                # part 2: brute force :)
                print("Part 2: " +  str(i*j*k))
                threeFound = True

# At the end of the day, it's much faster to write a solution that is fully O(n^2) or O(n^3) than
# to write one that has j/k starting from i. Too hard to deal with indexing when this is so easy